### PR TITLE
Update Cargo dependencies, add shim into Cargo workspace

### PR DESCRIPTION
**Pull request by Brooooooklyn**
- [x] upgrade dependencies
- [x] merge shim into the `cargo` workspace
- [x] make `cgo` work on Windows
- [ ] fix `cargo test`

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -27,6 +27,7 @@ members = [
   "crates/turbopack-static",
   "crates/turbopack-swc-utils",
   "crates/turbopack",
+  "shim",
```
Comment (NicholasLYang) [inline Cargo.toml:30]: There's an exclude rule for the shim that we'll need to remove too

File: crates/next-core/src/nodejs/mod.rs [hunk 2]
```diff
@@ -287,7 +287,7 @@ async fn render_static(
     let mut operation = match pool.operation().await {
         Ok(operation) => operation,
         Err(err) => {
-            return Ok(static_error(path, err, None, fallback_page).await?);
+            return static_error(path, err, None, fallback_page).await;
```
Comment (NicholasLYang) [inline crates/next-core/src/nodejs/mod.rs:290]: Any reason for these changes? Clippy?

**Review by NicholasLYang — CHANGES_REQUESTED** [review]

File: crates/next-core/src/nodejs/mod.rs (hunk 2 above)
Comment (Brooooooklyn) [reply crates/next-core/src/nodejs/mod.rs:290]: yes, clippy fixed it

File: shim/build.rs [hunk 3]
```diff
@@ -43,20 +43,21 @@ fn expect_release_lib() -> String {
...
+    // format is ${BUILD_ID}_${OS}_${ARCH}. Build id is, for goreleaser reasons,
+    // turbo-${OS}
     dir.push(format!("turbo-{platform}_{platform}_{arch}"));
     dir.push("lib");
     dir.to_string_lossy().to_string()
 }
 
 fn build_debug_libturbo() -> String {
-    let cli_path = "../cli";
+    let cli_path = env::current_dir().unwrap().parent().unwrap().join("cli");
```
Comment (jridgewell) [inline shim/build.rs:54]: Should this be using [CARGO_WORKSPACE_DIR](https://github.com/vercel/turbo/blob/1628df924e8e4d57046dbb00f325769cf808ec6d/.cargo/config.toml#L2) instead?

**Review by jridgewell — APPROVED** [review]

File: crates/turbopack-cli-utils/src/issue.rs [hunk 4]
```diff
@@ -52,8 +52,8 @@ impl clap::ValueEnum for IssueSeverityCliOption {
         &VARIANTS
     }
 
-    fn to_possible_value<'a>(&self) -> Option<clap::PossibleValue<'a>> {
-        Some(clap::PossibleValue::new(self.0.as_str()).help(self.0.as_help_str()))
+    fn to_possible_value<'a>(&self) -> Option<clap::builder::PossibleValue> {
```
[the remark this answers is not in the document]
Comment (Brooooooklyn) [reply crates/turbopack-cli-utils/src/issue.rs:55]: it because the `clap` was upgraded

**Review by jridgewell — APPROVED** [review]

**Comment by Brooooooklyn** [comment]
not sure why `turbo` tests failed on CI, @NicholasLYang can you take a look?

**Comment by NicholasLYang** [comment]
Ah, clap changed its help output in v4, so bumping it broke some tests. Also adding shim to the cargo workspace broke a couple things. Fixing it....

**Pull request closed without merging**